### PR TITLE
ShipHero: Added new action and source components

### LINKED ISSUE #5300: ShipHero
### App
ShipHero
https://developer.shiphero.com/

### PR DESCRIPTION
**Pull request by jcortes**
## WHY

Resolves #5300

## Summary by CodeRabbit

- **New Features**
    - Introduced actions to retrieve, update, and list orders and products via the ShipHero API.
    - Added webhook functionality for managing webhooks and handling order allocation, deallocation, and packing events.

- **Documentation**
    - Enhanced documentation for new actions and events, improving user guidance on API interactions.

- **Chores**
    - Updated package dependencies and versioning for improved functionality.

### CODE REVIEW

File: components/shiphero/package.json [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
   "description": "Pipedream ShipHero Components",
   "main": "dist/app/shiphero.app.mjs",
   "keywords": [
     "pipedream",
     "shiphero"
   ],
-  "files": ["dist"],
+  "files": [
+    "dist"
+  ],
```
Comment (luancazarine) [inline components/shiphero/package.json:12]:
```suggestion
```

**Review by luancazarine — CHANGES_REQUESTED** [review]
Hi @jcortes, I just added some minor suggestions!

File: components/shiphero/sources/order-allocated-instant/order-allocated-instant.mjs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import common from "../common/webhook.mjs";
+import events from "../common/events.mjs";
+import sampleEmit from "./test-event.mjs";
+
+export default {
+  ...common,
+  key: "shiphero_order-allocated-instant",
```
Comment (michelle0927) [inline components/shiphero/sources/order-allocated-instant/order-allocated-instant.mjs:7]:
```suggestion
  key: "shiphero-order-allocated-instant",
```

File: components/shiphero/sources/order-allocated-instant/order-allocated-instant.mjs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  version: "0.0.1",
+  dedupe: "unique",
+  methods: {
+    ...common.methods,
+    getEventName() {
+      return events.ORDER_ALLOCATED;
+    },
+    generateMeta(resource) {
+      return {
+        id: resource.id,
```
Comment (michelle0927) [inline components/shiphero/sources/order-allocated-instant/order-allocated-instant.mjs:20]: Should this be `resource.order_id`? The test-event doesn't show `id` in the resource.
```suggestion
        id: resource.order_id,
```

File: components/shiphero/sources/order-allocated-instant/order-allocated-instant.mjs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        summary: `Order Allocated: ${resource.order_id}`,
+        ts: Date.now(),
```
Comment (michelle0927) [inline components/shiphero/sources/order-allocated-instant/order-allocated-instant.mjs:22]: Optional: Couldn't this be `allocated_at`?
```suggestion
        ts: Date.parse(resource.allocated_at),
```

File: components/shiphero/sources/order-deallocated-instant/order-deallocated-instant.mjs [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import common from "../common/webhook.mjs";
+import events from "../common/events.mjs";
+import sampleEmit from "./test-event.mjs";
+
+export default {
+  ...common,
+  key: "shiphero_order-deallocated-instant",
```
Comment (michelle0927) [inline components/shiphero/sources/order-deallocated-instant/order-deallocated-instant.mjs:7]:
```suggestion
  key: "shiphero-order-deallocated-instant",
```

File: components/shiphero/sources/order-deallocated-instant/order-deallocated-instant.mjs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  version: "0.0.1",
+  dedupe: "unique",
+  methods: {
+    ...common.methods,
+    getEventName() {
+      return events.ORDER_DEALLOCATED;
+    },
+    generateMeta(resource) {
+      return {
+        id: resource.id,
```
Comment (michelle0927) [inline components/shiphero/sources/order-deallocated-instant/order-deallocated-instant.mjs:20]: Same comment as above.
```suggestion
        id: resource.order_id,
```

File: components/shiphero/sources/order-deallocated-instant/order-deallocated-instant.mjs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        summary: `Order Deallocated: ${resource.order_id}`,
+        ts: Date.now(),
```
Comment (michelle0927) [inline components/shiphero/sources/order-deallocated-instant/order-deallocated-instant.mjs:22]: Optional: Could use deallocated_at?
```suggestion
        ts: Date.parse(resource.deallocated_at),
```

File: components/shiphero/sources/order-packed-out-instant/order-packed-out-instant.mjs [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import common from "../common/webhook.mjs";
+import events from "../common/events.mjs";
+import sampleEmit from "./test-event.mjs";
+
+export default {
+  ...common,
+  key: "shiphero_order-packed-out-instant",
```
Comment (michelle0927) [inline components/shiphero/sources/order-packed-out-instant/order-packed-out-instant.mjs:7]:
```suggestion
  key: "shiphero-order-packed-out-instant",
```

File: components/shiphero/sources/order-packed-out-instant/order-packed-out-instant.mjs [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  version: "0.0.1",
+  dedupe: "unique",
+  methods: {
+    ...common.methods,
+    getEventName() {
+      return events.ORDER_PACKED_OUT;
+    },
+    generateMeta(resource) {
+      return {
+        id: resource.id,
```
Comment (michelle0927) [inline components/shiphero/sources/order-packed-out-instant/order-packed-out-instant.mjs:20]:
```suggestion
        id: resource.order_id,
```

**Review by michelle0927 — CHANGES_REQUESTED** [review]

**Review by GTFalcao — APPROVED** [review]
LGTM!

**Review by michelle0927 — APPROVED** [review]

**Pull request merged**